### ISSUE TITLE
Tidy fails if html contains a section <![endif]-->

### ISSUE THREAD

**Issue by foxinushka** [htacg/tidy-html5#153]
Email with bad html:
https://www.dropbox.com/s/pjo2ok9eyyjilfv/bad_html.mbox?dl=0

Tidy cuts all content bellow line:
```
D%252Fb%252F%253Fnode%253D4816518031">Amazon Mobile App</a></span></td>
```

**Comment by geoffmcl** [comment]
Thanks for the report, but I do not exactly understand what I am supposed to do with the dropbox posted bad_html.mbox!

It contains a whole mess of text... I tried extracting an xhtml... like
````
<!doctype html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
... down to ...
<!-- End Wrapper -->

<img src="http://www.amazon.co.uk/gp/r.html?R=163YAZTOO5PF7&C=1DJAGLDKFDR0W&H=DF24H6X4JQLHMM4KWXQ7RZLK2N0A&T=E&U=http%3A%2F%2Fimages-eu.amazon.com%2Fimages%2FG%2F01%2Fnav%2Ftransp.gif" /></body>
</html>
````

And, at first, tidy seemed to have no problem with this quite messy xhtml block... some 45 warnings, the last few were -
````
...
line 188 column 251 - Warning: unescaped & or unknown entity "&T"
line 188 column 255 - Warning: unescaped & or unknown entity "&U"
line 179 column 86 - Warning: missing </table>
line 172 column 44 - Warning: missing </table>
line 167 column 1 - Warning: missing </table>
line 163 column 1 - Warning: missing </table>
line 140 column 1 - Warning: missing </div>
line 137 column 1 - Warning: missing </table>
line 134 column 1 - Warning: <body> proprietary attribute "yahoo"
line 134 column 53 - Warning: <img> lacks "alt" attribute
line 137 column 1 - Warning: <table> proprietary attribute "yahoo
line 140 column 1 - Warning: <div> proprietary attribute "width"
line 169 column 55 - Warning: <p> proprietary attribute "width"
45 warnings, 0 errors were found!

Info: Document content looks like XHTML5
````

The sequence D%252Fb%252F%253Fnode%253D4816518031 is in the input 4 times, and it is agreed Tidy does seem to truncate the xhtml output after the line -
````
D%252Fb%252F%253Fnode%253D4816518031">Amazon Mobile App</a></span></td>
````

BUT so does the last cvs release of tidy, 2009, so this is a long, LONG term problem!

I will try some tests and see if I can find this long term problem, but it is QUITE messy and difficult using such a large sample, some 60K of data... and it does seem that it maybe due to comment opening and closing... but not sure yet...

It would really help if you could extract and construct a simple xhtml sample where tidy fails?

Show tidy version used, --version, input xhtml, configuration used, output received, and then output expected. 

Thanks...

**Comment by foxinushka** [comment]
Thanks!
I will post smaller example later.

**Comment by geoffmcl** [comment]
Thank you. That would be really helpful... more examples ALWAYS help...

But I have found the problem. AND have constructed a simple sample -
````
<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<title>Tidy fails if html contains a section <![endif]--> #153</title>
</head>
<body>
<p>Para 1 - before if</p>
<!--[if !mso 9]><!-->
<p>Para 2 in if</p>
<![endif]-->
<p>Para 3 - tidy missed paragraph</p>
</body>
</html>
````

All tidy, probably since it was first coded to handle secions, maybe sometime well before 2009, will see the 'endif' and 100% expects `']>'`, not `]-->`!

So will continue to march to the end of the file seeking the `]>` it expects, so EVERY thing after this section is discarded.

I have coded a partial fix, so that it now reads and handles the rest of the file, but the pprint.c module does not yet output that section correctly.

But now running tidy on your big sample creates an output that displays similar/same as the original, which is always ridy's aim.

Hope you do not mind that I have amended your title to more suit what was found.

Look forward to your further report after testing this 4.9.10 version.
